1.fee_bumping/fee_bumping.md

```diff
@@ -153,25 +153,217 @@ Bitcoin blockchain, such as exchanges or custodians:
+is lower.
+
+- implication 1: if a wallet runs out of confirmed transactions (ie all of its
+  coins are tied up in unconfirmed transactions), then it can't use the output
+  of some of those transactions as additional inputs to bump the fee on its other
+  transactions.
 
-### 5 conditions for RBF
-
-TODO
+> 3. The replacement transaction pays an absolute fee of at least the sum paid
+> by the original transactions.
+
+- implication 1: if outputs from the original transaction have been spent, then
+  the replacement transaction must exceed both the *feerate* and *total fee* of
+  the original transaction and its descendants. If the child transaction is large,
+  or there are many descendant transactions, then the total fee required to
+  replace the original transaction could be very large. This is known as 
+  *transaction pinning*.
+
+> 4. The replacement transaction must also pay for its own bandwidth at or
+> above the rate set by the node's minimum relay fee setting. For example,
+> if the minimum relay fee is 1 satoshi/byte and the replacement transaction
+> is 500 bytes total, then the replacement must pay a fee at least 500
+> satoshis higher than the sum of the originals.
+
+Rationale: this is an anti-DoS measure to prevent an adversary from repeatedly
+sending replacement transactions with miniscule feebumps and making the peer
+revalidate many versions of the same transaction.
+
+- implication 1: for transactions at low feerates, the proportional change
+  in feerate from bumping is large. A transaction with a feerate of
+  1 sat/byte needs to *double* its fee to satisfy this rule. For very
+  large transactions (eg large consolidations), then the amount of money
+  spent on fees could be high.
+
+> 5. The number of original transactions to be replaced and their descendant
+> transactions which will be evicted from the mempool must not exceed a
+> total of 100 transactions.
+
+Rationale: this is an anti-DoS measure to prevent an adversary from sending
+replacement transactions that cause their peer to walk through many
+transactions in their mempool. This scan through the mempool needs to be done
+before calculating whether the replacement transaction has enough fee to
+replace the original transactions, so an adversary could cause the peer to walk
+the mempool without causing transactions to be replaced, which would be a free
+attack.
+
+- implication 1: A transaction can be pinned by having > 100 descendants (see
+  rule 3 for the definition of pinning). Note that if you can sign for some of
+  those 100 descendants, you can RBF in multiple steps. For example, you could
+  replace 50 of the descendant transactions with replacement transaction A, and
+  then replace the original ancestor transaction with replacement transaction
+  B. Because replacement transaction A cleared 50 of the descendant transactions
+  out of the mempool, replacement transaction B is only replacing 51 transactions
+  and will be accepted.
```

Comment:
It might be worth noting that anyone doing two-step (or multi-step) replacements like this should probably leave at least several seconds (perhaps more) between sending the updates so that the first update has reached almost all nodes before the second update starts propagating.  Otherwise, there's a chance nodes will receive the second update first, reject it, cache that txid as rejected, and then refuse to reconsider that txid even after receiving the first update (until the rejection cache is cleared, which I think is at each block in Bitcoin Core).

Reply:
@dgpv has pointed out that this cannot be the case when you're only replacing one package of transactions due to the descendant limit: https://github.com/bitcoinops/scaling-book/pull/5/files#r238054439 so I've removed this implication entirely.